SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -65,27 +108,49 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
+	DelegationMap.Add(SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID, AuthoritativeClientPartitionId);
 
-	for (auto& SubobjectInfoPair : Info.SubobjectInfo)
+	// Add debugging utilities, if we are not compiling a shipping build
+#if !UE_BUILD_SHIPPING
+	if (NetDriver->SpatialDebugger != nullptr)
 	{
-		const FClassInfo& SubobjectInfo = SubobjectInfoPair.Value.Get();
+		check(NetDriver->VirtualWorkerTranslator != nullptr);
 
-		// Static subobjects aren't guaranteed to exist on actor instances, check they are present before adding to delegation component
-		TWeakObjectPtr<UObject> Subobject = PackageMap->GetObjectFromUnrealObjectRef(FUnrealObjectRef(EntityId, SubobjectInfoPair.Key));
-		if (!Subobject.IsValid())
-		{
-			continue;
-		}
+		const PhysicalWorkerName* PhysicalWorkerName =
+			NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(IntendedVirtualWorkerId);
+		FColor InvalidServerTintColor = NetDriver->SpatialDebugger->InvalidServerTintColor;
+		FColor IntentColor =
+			PhysicalWorkerName != nullptr ? SpatialGDK::GetColorForWorkerName(*PhysicalWorkerName) : InvalidServerTintColor;
+
+		const bool bIsLocked = NetDriver->LockingPolicy->IsLocked(Actor);
+
+		SpatialDebugging DebuggingInfo(SpatialConstants::INVALID_VIRTUAL_WORKER_ID, InvalidServerTintColor, IntendedVirtualWorkerId,
+									   IntentColor, bIsLocked);
+		ComponentDatas.Add(DebuggingInfo.CreateComponentData());
 	}
+#endif // !UE_BUILD_SHIPPING
 
-	// We want to have a stably named ref if this is an Actor placed in the world.
-	// We use this to indicate if a new Actor should be created, or to link a pre-existing Actor when receiving an AddEntityOp.
-	// Previously, IsFullNameStableForNetworking was used but this was only true if bNetLoadOnClient=true.
-	// Actors with bNetLoadOnClient=false also need a StablyNamedObjectRef for linking in the case of loading from a snapshot or the server
-	// crashes and restarts.
-	TSchemaOption<FUnrealObjectRef> StablyNamedObjectRef;
-	TSchemaOption<bool> bNetStartup;
+	// Add actual load balancing components
+	ComponentDatas.Add(NetOwningClientWorker(AuthoritativeClientPartitionId).CreateComponentData());
+	ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(IntendedVirtualWorkerId));
+	ComponentDatas.Add(AuthorityDelegation(DelegationMap).CreateComponentData());
+}
+
+void EntityFactory::WriteUnrealComponents(TArray<FWorkerComponentData>& ComponentDatas, USpatialActorChannel* Channel,
+										  uint32& OutBytesWritten)
+{
+	AActor* Actor = Channel->Actor;
+	UClass* Class = Actor->GetClass();
+	Worker_EntityId EntityId = Channel->GetEntityId();
+
+	const FClassInfo& Info = ClassInfoManager->GetOrCreateClassInfoByClass(Class);
+
+	Worker_ComponentId ActorInterestComponentId = ClassInfoManager->ComputeActorInterestComponentId(Actor);
+	const USpatialGDKSettings* SpatialSettings = GetDefault<USpatialGDKSettings>();
+
+	// Leaving this code block here to guarantee the resolution of outers of stably named actors
```

Comment:
Hello @improbable-valentyn, this is the code we discussed.
I left in the actor resolution, and I also experimentally left in a slow check for whether the stably named object refs constructed using the new method are the same as the old ones.
Now, the slow check I have not seen fire in my small testing, and also commenting out the resolution of outers didn't seem to affect the game either.